### ISSUE TITLE
All patches except for `crt` and `no-background` work on Balatro Version 1.0.0n-FULL

### ISSUE THREAD

**Issue by Ron-J1** [antipatico/balatromobile#6]
Hey,

thanks for porting this to Python! Much appreciated. 

I wanted to report that with the current Balatro Version on Steam (1.0.0n-FULL) all patches except for `crt` and `no-background` work.

Those last two both raise the following error:

```python-traceback
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "...\envs\balatro\Scripts\balatromobile.exe\__main__.py", line 7, in <module>
  File "...\envs\balatro\Lib\site-packages\balatromobile\gamble.py", line 21, in main
    android(args)
  File "...\envs\balatro\Lib\site-packages\balatromobile\gamble.py", line 43, in android
    patch.apply(balatro, balatro_version)
  File "...\envs\balatro\Lib\site-packages\balatromobile\patcher.py", line 79, in apply
    p.apply_all(balatro)
  File "...\envs\balatro\Lib\site-packages\balatromobile\patcher.py", line 48, in apply_all
    [p.apply_all(balatro) for p in self.patch_files]
  File "...\envs\balatro\Lib\site-packages\balatromobile\patcher.py", line 48, in <listcomp>
    [p.apply_all(balatro) for p in self.patch_files]
     ^^^^^^^^^^^^^^^^^^^^
  File "...\envs\balatro\Lib\site-packages\balatromobile\patcher.py", line 32, in apply_all
    [p.apply(balatro / self.target_file) for p in self.patches]
  File "...\envs\balatro\Lib\site-packages\balatromobile\patcher.py", line 32, in <listcomp>
    [p.apply(balatro / self.target_file) for p in self.patches]
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "...\envs\balatro\Lib\site-packages\balatromobile\patcher.py", line 22, in apply
    patched = "\n".join([l if self.search_string not in l else self.content for l in target.read_text().splitlines()])
                                                                                     ^^^^^^^^^^^^^^^^^^
  File "...\envs\balatro\Lib\pathlib.py", line 1059, in read_text
    return f.read()
           ^^^^^^^^
  File "...\envs\balatro\Lib\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'charmap' codec can't decode byte 0x8f in position 111728: character maps to <undefined>
```

Well, that is, the others work after modifying the version check to always return `True` ;)

**Comment by antipatico** [comment]
Thanks for reporting. If you provide me the `depot id` and `manifest id` of the build you are testing using [SteamDB](https://steamdb.info/app/2379780/) I will download it and try to replicate the bug on my machine. If you find a solution, feel free to contribute by opening a PR.

**Comment by ronald-jaepel** [comment]
I think those are:

Depot ID: 2379781
Build ID: 13682455
Manifest ID: 1435140510430378530

I'm perfectly happy without these two patches, just thought I'd share what I've found.

**Comment by antipatico** [comment]
Do you experience the same bug in supported game versions? (1.0.1c-FULL and 1.0.1e-FULL as of today)

**Comment by antipatico** [comment]
@Ron-J1 I tested on my linux laptop with version `1.0.0n-FULL` of the game and version `0.3.1` of this project and I cannot reproduce.
After adding `1.0.0n-FULL` to the list of supported game version on [basic.toml#7](https://github.com/antipatico/balatromobile/blob/4e986fdead90b10282ad49081da2cd118d137a7f/balatromobile/patches/basic.toml#L7) I run the following:

```bash
balatromobile android Balatro-13682455-1.0.1n-FULL.exe --patches basic,crt,no-background
```

And the script works without throwing any error. I suspect the problem may be in some Windows specific behavior related to pathlib. If I have to guess, I would say that having forward slash in the `target_file` variable inside the `crt.toml` makes Window sad. If that is the case, why `no-background` would be broken on Windows would be a mystery to me.

Unfortunately, without further input from Windows users, I will not be able to tackle this one down.

EDIT: it may also be related on which order you supplied the patches through the command line. The order in which you supply them is the order in which they are applied. It would make sense that some patches may become incompatible if a patch is run before another. This is a shot in the dark though, since I did not investigate in this behavior.

**Comment by ronald-jaepel** [comment]
> Do you experience the same bug in supported game versions? (1.0.1c-FULL and 1.0.1e-FULL as of today)

How do I find the manifest IDs for those versions? I can't find them on SteamDB.

**Comment by antipatico** [comment]
> > Do you experience the same bug in supported game versions? (1.0.1c-FULL and 1.0.1e-FULL as of today)
> 
> How do I find the manifest IDs for those versions? I can't find them on SteamDB.

You have to look for the [public-experimental branch](https://steamdb.info/app/2379780/depots/?branch=public_experimental) 

1.0.1c-FULL is build 13990650 having depot id `2379781` manifest id `6700732020928482327`
1.0.1e-FULL is build 14188887 having depot id `2379781` and manifest id `7501048753131013242`

**Comment by ronald-jaepel** [comment]
Great, thanks! I looked into the error more and it looks like it's struggling to decode Chinese and Japanese characters in the `game.lua` file. After changing the encoding for the `.read_text()` to `.read_text(encoding="utf-8")` it now works for both patches on `1.0.0n`. I'll try the other Balatro versions next.

**Comment by ronald-jaepel** [comment]
I've openend PR #9

**Comment by ronald-jaepel** [comment]
`1.0.1c-FULL` and `1.0.1e-FULL` both work from my PR.

**Comment by antipatico** [comment]
Should be fixed by #9 .
Thanks @ronald-jaepel  for the PR
